kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java

```diff
@@ -42,7 +42,7 @@
 /**
  * Base class for goals which deploy the generated artifacts into the Kubernetes cluster
  */
-@Mojo(name = "apply", requiresDependencyResolution = ResolutionScope.COMPILE, defaultPhase = LifecyclePhase.INSTALL)
+@Mojo(name = "apply", requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, defaultPhase = LifecyclePhase.INSTALL)
```

Comment:
Same change should be applicable to every child:
- OpenshiftApplyMojo
- DebugMojo
- OpenshiftDebugMojo
- DeployMojo
- OpenshiftDeployMojo
- LogMojo
- OpenshiftLogMojo